### PR TITLE
Delete `setup.py` (Move to hatchling as a build-backend)

### PR DESCRIPTION
**Pull request by AlexWaygood**
I used the following `pyproject.toml` files as templates:
- https://github.com/PyCQA/flake8-bugbear/blob/main/pyproject.toml
- https://github.com/psf/black/blob/main/pyproject.toml
- https://github.com/AlexWaygood/typeshed-stats/blob/main/pyproject.toml

AFAICT, everything seems to be working as expected locally. But, please review with care, as I'm _hardly_ a packaging expert :)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,69 @@
...
+authors = [
+  { name="Łukasz Langa", email="=lukasz@langa.pl" },
+]
+maintainers = [
+  { name="Jelle Zijlstra", email="jelle.zijlstra@gmail.com" },
+  { name="Alex Waygood", email="alex.waygood@gmail.com" },
+  { name="Sebastian Rittau" },
+  { name="Akuli" },
+  { name="Shantanu" },
+]
```
Comment (AlexWaygood) [inline pyproject.toml:17]: I added this section

File: pyproject.toml [hunk 2]
```diff
@@ -1,3 +1,69 @@
...
+keywords = [
+    "flake8",
+    "pyi",
+    "bugs",
+    "pyflakes",
+    "linter",
+    "qa",
+    "stubs",
+    "typing",
+]
```
Comment (AlexWaygood) [inline pyproject.toml:31]: I removed "pylint" from this section and added "stubs" and "typing"

File: pyproject.toml [hunk 3]
```diff
@@ -1,3 +1,69 @@
...
+classifiers = [
+    "Development Status :: 3 - Alpha",
```
Comment (AlexWaygood) [inline pyproject.toml:33]: I kept this as it was, but we could maybe consider bumping it to "beta".

File: pyproject.toml [hunk 4]
```diff
@@ -1,3 +1,69 @@
...
+    "flake8 >= 3.2.1, < 7.0.0",
+    "pyflakes >= 2.1.1",
+    "ast-decompiler >= 0.7.0, < 1.0; python_version < '3.9'",
+]
+
+[project.urls]
+"Homepage" = "https://github.com/PyCQA/flake8-pyi"
+"Source" = "https://github.com/PyCQA/flake8-pyi"
+"Bug Tracker" = "https://github.com/PyCQA/flake8-pyi/issues"
+"Changelog" = "https://github.com/PyCQA/flake8-pyi/blob/master/CHANGELOG.md"
```
Comment (AlexWaygood) [inline pyproject.toml:59]: I added a few more links here

**Review by AlexWaygood — COMMENTED** [review]
Here's the things that I changed a little from our existing `setup.py` file:

File: pyproject.toml (hunk 3 above)
Comment (JelleZijlstra) [reply pyproject.toml:33]: Sure. Personally I don't get anything useful out of this classifier.

**Review by JelleZijlstra — APPROVED** [review]

**Comment by AlexWaygood** [comment]
🥳

**Pull request merged**